Improvements to CSS

danielhaim1 opened the pull request:
This is a 🔦 documentation change.

## Summary

General improvements to the stylesheet used for Jekyll documentations.
- I've added the correct font features for `Lato`
- Added `text-rendering` option
- Improved the `font-` rules (fall backs)
- ~~Improved order of CSS properties~~

Tested on multiple pages.

ashmaroli commented:
Hello @danielhaim1, thank you for submitting this PR.
While we don't *enforce* a linter on the documentation site's CSS, and as such don't have rules in place,
we *prefer* indenting with just **2 spaces per level** instead of **4 spaces or more per level**

danielhaim1 commented:
- Changed indenting to `2` instead of `4` spaces
- Added `font-display` feature to `FontAwesome`

ashmaroli commented:
I see a lot of unnecessary changes like changing double-quotes to single-quotes, removal of comments and other whitespace changes.

Please limit the amount of changes to just those that are actually necessary.

danielhaim1 commented:
@ashmaroli thanks for reviewing again.

I updated `docs/_sass/_fonts.scss` and improved the comments there. The original comments were removed because they were incorrect. Lato (300) is `Lato Light`,  but stated as `lato-300 latin`

In other cases, I converted inline CSS comments (`/* comment */` to `// comment`) which are better for `scss`.

In other changes, I've sorted the CSS properties with each selector declaration in a predefined order to be as close as possible to the browser engine.

ashmaroli commented:
You're welcome. Understandably, the amount of changes increases the amount of time for reviewing.

You can help us out by going through the `diff` and reverting changes to quotes, undoing removal of blank lines at end of file and basically revert all whitespace changes that will keep the diff shorter

danielhaim1 commented:
@ashmaroli reverting the files back to normal, and only added my most critical changes.
spaces are now back to normal, and the file is no longer sorted by property name.

danielhaim1 commented:
Let me know if you'd like me to comb the file again, it saves approx ~300 lines.

ashmaroli commented:
The most noticeable change from this PR is the rendering of numerals:

## Current `master`:

![master](https://user-images.githubusercontent.com/12479464/65830802-7aa1a680-e2d0-11e9-8eab-8c8ed0cb498b.png)

## This PR in current state:

![pull_request](https://user-images.githubusercontent.com/12479464/65830812-9147fd80-e2d0-11e9-977a-3150f85253e4.png)

devmoses commented:
![current: master](https://user-images.githubusercontent.com/12479464/65830802-7aa1a680-e2d0-11e9-8eab-8c8ed0cb498b.png)

Yup! The reverted looks good to me as well.

ashmaroli commented:
> The reverted looks good

@devmoses Um.. I don't see the revert commit here.
The last preview-build on this branch still shows the numerals rendered with lesser vertical height:
https://deploy-preview-7834--jekyllrb.netlify.com/news/2019/08/20/jekyll-4-0-0-released/

DirtyF reviewed (approved):
This looks good to me, thanks, and sorry for the late review :)

DirtyF commented:
@jekyll: merge +site
